ipykernel/mplshim.py

```diff
@@ -0,0 +1,89 @@
+"""Import hooks to activate our inline backend when matplotlib is loaded
+"""
+
+import imp
+import sys
+
+def set_inline_backend(mpl):
+    from IPython.core.pylabtools import backends, configure_inline_support
+    from .kernelapp import IPKernelApp
+    mpl.interactive(True)
+    mpl.use(backends['inline'])
+    configure_inline_support(IPKernelApp.instance().shell, backends['inline'])
+
+class Py2MPLFinderShim(object):
+    @classmethod
+    def find_module(cls, fullname, path=None):
+        if (fullname != 'matplotlib') or (path is not None):
```

Comment:
There was also discussion at scipy with @weathergod to delay actually forcing the back end until the first figure is created.